lib/util/platform.js

```diff
@@ -354,6 +355,23 @@ shaka.util.Platform = class {
     return shaka.util.Platform.userAgentContains_('Sky_STB');
   }
 
+  /**
+   * Check if the current platform is XOne.
+   */
+  static isXOne() {
+    let rdk = false;
+    if (window && Object.keys(window).includes('ServiceManager')) {
```

Comment:
Does this simpler version work instead?

```js
01234567890123456789012345678901234567890123456789012345678901234567890123456789
if (window.ServiceManager && window.ServiceManager.getServiceForJavaScript &&
    navigator.platform.includes('Linux aarch64') &&
    shaka.util.Platform.userAgentContains_('AppleWebKit') &&
    shaka.util.Platform.userAgentContains_('Version/8.0')) {
```

You will need to add a file like `externs/xbox.js` to satisfy the compiler that `ServiceManager` and `getServiceForJavaScript` might exist.  The types can be very simple if you don't need to call into any of that functionality.

Reply:
Thanks i did it and also refactor the previous check